Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/scheduleSettingsContent.jsx

```diff
@@ -0,0 +1,329 @@
+        };
+
+        this.batchTypeToDisplayMap = {};
+        Models.batchTypes.forEach(batchType => {
+            this.batchTypeToDisplayMap[batchType.key] = batchType.name;
+        });
+    }
+
+    render() {
+        return (
+            <div style={rootStyle}>
+                <StatementBox
+                    icon="SignOut"
+                    statement="Schedule recurring and/or one time batch jobs. For the jobs scheduled correctly, the scheduler will create and start jobs."
+                />
+                {this.renderContent()}
+            </div>
+        );
+    }
+
+    renderContent() {
+        return (
+            <div style={contentStyle}>
+                {this.renderLeftPane()}
+                {this.renderRightPane()}
+            </div>
+        );
+    }
+
+    renderLeftPane() {
+        this.showSelectionInDetailsList();
+
+        const batchList = this.props.batchList;
+        batchList.forEach(item => {
+            item.typeDisplay = this.batchTypeToDisplayMap[item.type];
+            item.status = !item.disabled ? "Active" : "Disabled";
+        });
+
+        return (
+            <div style={leftPaneStyle}>
+                <PanelHeader style={Styles.panelHeaderStyle}>Schedule Batch Jobs</PanelHeader>
+                <PanelHeaderButtons style={panelHeaderButtonStyle}>{this.renderButtons()}</PanelHeaderButtons>
+
+                <div style={leftPaneContentStyle}>
+                    <div style={listContainerStyle}>
+                        <DetailsList
+                            items={batchList}
+                            columns={batchColumns}
+                            onActiveItemChanged={(item, index) => this.onSelectBatchItem(item, index)}
+                            setKey="batchList"
+                            layoutMode={DetailsListLayoutMode.justified}
+                            selectionMode={SelectionMode.single}
+                            selectionPreservedOnEmptyClick={true}
+                            isHeaderVisible={true}
+                            checkboxVisibility={CheckboxVisibility.hidden}
+                            selection={this.batchSelection}
+                        />
+                    </div>
+                </div>
+            </div>
+        );
+    }
+
+    renderRightPane() {
+        return (
+            <div style={rightPaneStyle}>
+                <PanelHeader style={Styles.panelHeaderStyle}>Settings</PanelHeader>
+                <ScrollableContentPane backgroundColor={Colors.neutralLighterAlt}>{this.renderBatchSettings()}</ScrollableContentPane>
+            </div>
+        );
+    }
+
+    renderButtons() {
+        return [this.renderAddBatchButton(), this.renderDeleteBatchButton()];
+    }
+
+    renderAddBatchButton() {
+        let recurringExists = this.props.batchList.filter(b => b.type  === Models.batchTypeEnum.recurring);
+
+        const menuItems = (recurringExists === undefined || recurringExists.length < 1)
```

Comment:
FYI: recurringExists === undefined is equivalent to !recurringExists. I recommend to use the latter.

Reply:
done